backend/src/nodes/image_util_nodes.py

```diff
@@ -5,80 +5,98 @@
 
 import cv2
 import numpy as np
+import numpy.typing as npt
 from sanic.log import logger
 
 from .categories import IMAGE_UTILITY
 from .node_base import NodeBase
 from .node_factory import NodeFactory
 from .properties.inputs import *
 from .properties.outputs import *
-from .utils.image_utils import alpha_overlay, calculate_ssim
+from .utils.image_utils import blend_images, calculate_ssim
 from .utils.pil_utils import *
 from .utils.utils import get_h_w_c
 
+ndarray32 = npt.NDArray[np.float32]
 
-@NodeFactory.register("chainner:image:overlay")
-class ImOverlay(NodeBase):
-    """OpenCV transparency overlay node"""
+
+@NodeFactory.register("chainner:image:blend")
+class ImBlend(NodeBase):
+    """Blending mode node"""
 
     def __init__(self):
         """Constructor"""
         super().__init__()
-        self.description = "Overlay transparent images on base image."
+        self.description = """Blends overlay image onto base image using 
+            specified mode and opacities."""
         self.inputs = [
-            ImageInput("Base").with_id(0),
+            ImageInput("Base"),
             SliderInput(
                 "Base Opacity",
                 maximum=100,
                 default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(5),
-            ImageInput("Overlay A").with_id(1),
-            SliderInput(
-                "Opacity A",
-                maximum=100,
-                default=50,
-                step=0.1,
-                controls_step=1,
-                unit="%",
-            ).with_id(2),
-            ImageInput("Overlay B").make_optional().with_id(3),
+            ),
+            ImageInput("Overlay").make_optional(),
             SliderInput(
-                "Opacity B",
+                "Overlay Opacity",
                 maximum=100,
-                default=50,
+                default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(4),
+            ),
+            DropDownInput(
+                "Blend Mode",
+                [
+                    {"option": "Normal", "value": 0},
+                    {"option": "Multiply", "value": 1},
+                    {"option": "Darken", "value": 2},
+                    {"option": "Lighten", "value": 3},
+                    {"option": "Add", "value": 4},
+                    {"option": "Color Burn", "value": 5},
+                    {"option": "Color Dodge", "value": 6},
+                    {"option": "Reflect", "value": 7},
+                    {"option": "Glow", "value": 8},
+                    {"option": "Overlay", "value": 9},
+                    {"option": "Difference", "value": 10},
+                    {"option": "Negation", "value": 11},
+                    {"option": "Screen", "value": 12},
+                    {"option": "Xor", "value": 13},
```

Comment:
Please make class constants (or similar) for these values. These integer values are essentially just magic numbers that just happen to line up with the numbers used by image blender. That's way too brittle.